From 378ef50bc121690ee564ca56e06bdb5b992c1c70 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Mon, 24 Dec 2018 15:56:53 +0800
Subject: [PATCH 444/706] ptp_qoriq: make eTSEC 1588 timer memory map specific

There will be three kinds of 1588 timer for ptp_qoriq
driver to support. They are 1588 timers for eTSEC, FMan
and ENETC. Actually FMan and ENETC use same memory map.
So let's make eTSEC 1588 timer memory map specific, and
support ENETC 1588 timer with same memory map with FMan.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
(cherry picked from commit 306dc8f3450e119ce56fffd34a4a8e7e225e85a8)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/ptp/ptp_qoriq.c       | 10 +++++-----
 include/linux/fsl/ptp_qoriq.h | 18 +++++++++---------
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/drivers/ptp/ptp_qoriq.c b/drivers/ptp/ptp_qoriq.c
index 224bf4593a7f..5dd7c5208edf 100644
--- a/drivers/ptp/ptp_qoriq.c
+++ b/drivers/ptp/ptp_qoriq.c
@@ -462,11 +462,11 @@ int qoriq_ptp_init(struct device *dev, struct qoriq_ptp *qoriq_ptp,
 	else
 		qoriq_ptp->little_endian = false;
 
-	if (of_device_is_compatible(node, "fsl,fman-ptp-timer")) {
-		qoriq_ptp->regs.ctrl_regs = base + FMAN_CTRL_REGS_OFFSET;
-		qoriq_ptp->regs.alarm_regs = base + FMAN_ALARM_REGS_OFFSET;
-		qoriq_ptp->regs.fiper_regs = base + FMAN_FIPER_REGS_OFFSET;
-		qoriq_ptp->regs.etts_regs = base + FMAN_ETTS_REGS_OFFSET;
+	if (of_device_is_compatible(node, "fsl,etsec-ptp")) {
+		qoriq_ptp->regs.ctrl_regs = base + ETSEC_CTRL_REGS_OFFSET;
+		qoriq_ptp->regs.alarm_regs = base + ETSEC_ALARM_REGS_OFFSET;
+		qoriq_ptp->regs.fiper_regs = base + ETSEC_FIPER_REGS_OFFSET;
+		qoriq_ptp->regs.etts_regs = base + ETSEC_ETTS_REGS_OFFSET;
 	} else {
 		qoriq_ptp->regs.ctrl_regs = base + CTRL_REGS_OFFSET;
 		qoriq_ptp->regs.alarm_regs = base + ALARM_REGS_OFFSET;
diff --git a/include/linux/fsl/ptp_qoriq.h b/include/linux/fsl/ptp_qoriq.h
index 499f9f4775d0..ed5d55f7d5de 100644
--- a/include/linux/fsl/ptp_qoriq.h
+++ b/include/linux/fsl/ptp_qoriq.h
@@ -57,15 +57,15 @@ struct qoriq_ptp_registers {
 };
 
 /* Offset definitions for the four register groups */
-#define CTRL_REGS_OFFSET	0x0
-#define ALARM_REGS_OFFSET	0x40
-#define FIPER_REGS_OFFSET	0x80
-#define ETTS_REGS_OFFSET	0xa0
-
-#define FMAN_CTRL_REGS_OFFSET	0x80
-#define FMAN_ALARM_REGS_OFFSET	0xb8
-#define FMAN_FIPER_REGS_OFFSET	0xd0
-#define FMAN_ETTS_REGS_OFFSET	0xe0
+#define ETSEC_CTRL_REGS_OFFSET	0x0
+#define ETSEC_ALARM_REGS_OFFSET	0x40
+#define ETSEC_FIPER_REGS_OFFSET	0x80
+#define ETSEC_ETTS_REGS_OFFSET	0xa0
+
+#define CTRL_REGS_OFFSET	0x80
+#define ALARM_REGS_OFFSET	0xb8
+#define FIPER_REGS_OFFSET	0xd0
+#define ETTS_REGS_OFFSET	0xe0
 
 
 /* Bit definitions for the TMR_CTRL register */
-- 
2.17.1

